cmake/OpenCVFindLAPACK.cmake

```diff
@@ -58,16 +58,52 @@ macro(ocv_lapack_check)
     string(REPLACE ";" "\n" _lapack_content "${_lapack_content}")
     ocv_update_file("${CBLAS_H_PROXY_PATH}" "${_lapack_content}")
 
+    if(CMAKE_GENERATOR MATCHES "Visual Studio"  # MSBuild
+        AND LAPACK_IMPL STREQUAL "MKL"
+        AND ";${LAPACK_LIBRARIES};" MATCHES ";tbb;" AND TARGET tbb
+        AND DEFINED TBB_INTERFACE_VERSION AND NOT (TBB_INTERFACE_VERSION LESS 12000)  # oneTBB/oneAPI workaround
+    )
+      # workaround DEFAULTLIB:tbb12.lib issue
+      get_target_property(_tbb_lib tbb IMPORTED_LOCATION)
+      if(NOT _tbb_lib)
+        get_target_property(_tbb_lib tbb IMPORTED_LOCATION_RELEASE)
+      endif()
+      if(_tbb_lib AND NOT OPENCV_SKIP_WORKAROUND_MKL_LINK_DIRECTORIES_TBB)
+        # MSBuild drops content of 'LIB' environment variable,
+        # so pass TBB library directory through `link_directories()`
+        get_filename_component(_tbb_lib_dir "${_tbb_lib}" DIRECTORY)
+        message(STATUS "MKL: adding '${_tbb_lib_dir}' to link directories (workaround DEFAULTLIB issue)")
+        link_directories("${_tbb_lib_dir}")
```

Comment:
Observations:
- direct invocation of link.exe/cl.exe works fine (with setvars.bat)
- invocation through MSBuild doesn't work

Reason: MSBuild drops/replaces some environment variables, including `LIB` which points to TBB's library directory (specified in setvars).